test/vcf/report_writer_test.cpp

```diff
@@ -0,0 +1,137 @@
+/**
+ * Copyright 2016 EMBL - European Bioinformatics Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <memory>
+#include <iostream>
+#include <fstream>
+
+#include <boost/filesystem.hpp>
+
+#include "catch/catch.hpp"
+#include "sqlite/sqlite3.h"
+
+#include "vcf/file_structure.hpp"
+#include "vcf/validator.hpp"
+#include "vcf/error.hpp"
+#include "vcf/report_writer.hpp"
+
+
+namespace ebi
+{
+  size_t const default_line_buffer_size = 64 * 1024;
+
+  template <typename Container>
+  std::istream & readline(std::istream & stream, Container & container)
+  {
+      char c;
+
+      container.clear();
+
+      do {
+          stream.get(c);
+          container.push_back(c);
+      } while (!stream.eof() && c != '\n');
+
+      return stream;
+  }
+
+  bool is_valid(std::string path, ebi::vcf::ReportWriter &output)
+  {
+      std::ifstream input{path};
+      if (!input) {
+          throw std::runtime_error("file not found: " + path);
+      }
+      auto source = ebi::vcf::Source{path, ebi::vcf::InputFormat::VCF_FILE_VCF, ebi::vcf::Version::v41};
+      auto records = std::vector<ebi::vcf::Record>{};
+
+      auto validator = ebi::vcf::FullValidator_v41{
+          std::make_shared<ebi::vcf::Source>(source),
+          std::make_shared<std::vector<ebi::vcf::Record>>(records)};
+
+      std::vector<char> line;
+      line.reserve(default_line_buffer_size);
+
+      while (readline(input, line)) {
+          validator.parse(line);
+          for (auto error : *validator.errors()) {
+              output.write_error(*error);
+          }
+          for (auto warn : *validator.warnings()) {
+              output.write_warning(*warn);
+          }
+      }
+
+      validator.end();
+
+      return validator.is_valid();
+  }
+
+  TEST_CASE("output errors to sqlite", "[output]")
```

Comment:
Before the integration test already implemented, it would be interesting to have unit tests that validate the methods from the writer interface in an isolated way: opening the database, writing an error/warning, etc.